server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -97,7 +97,22 @@ export function create(
                 crypto.randomBytes(4).toString("hex"),
                 values);
 
-            handleResponse(createP.then(() => id), response, undefined, 201);
+            // Generate creation token given a jwt from header
+            const authorizationHeader = request.header("Authorization");
+            const tokenRegex = /Basic (.+)/;
+            const tokenMatch = tokenRegex.exec(authorizationHeader);
+            const token = tokenMatch[1];
+
+            const tenantKeyP = tenantManager.getKey(tenantId);
+
+            handleResponse(Promise.all([createP, tenantKeyP]).then(([_, key]) => {
+                return generateToken
+                    ? {
+                        id,
+                        token: getCreationToken(token, key, id),
+                    }
+                    : id;
```

Comment:
For the sake of future simplicity here, could we have the generateToken==false case return `{ id: string}`? That way, we can eventually remove the `| string` handling in the driver, and it will simplify the potential for adding more properties to the response body here.

This change would then require token to be expected to be possibly undefined in the driver. A little more complicated now for less complicated eventually 😄